examples/hnas/abohb_example.py

```diff
@@ -0,0 +1,229 @@
+    learning_rate = args.learning_rate
+    wd = args.wd
+
+    # Make sure you downloaded the dataset before running AutoGluon. Downloading the data inside the
+    # objective function might lead to crashes of the python process (at least on MAC OS).
+    data_train = datasets.MNIST(root='data', train=True,
+                                download=False, transform=transforms.ToTensor())
+
+    # We use 50000 samples for training and 10000 samples for validation
+    indices = list(range(data_train.data.shape[0]))
+    train_idx, valid_idx = indices[:50000], indices[50000:]
+    train_sampler = SubsetRandomSampler(train_idx)
+    valid_sampler = SubsetRandomSampler(valid_idx)
+
+    train_loader = torch.utils.data.DataLoader(data_train, batch_size=batch_size,
+                                               sampler=train_sampler, drop_last=True)
+    valid_loader = torch.utils.data.DataLoader(data_train, batch_size=batch_size,
+                                               sampler=valid_sampler, drop_last=True)
+
+    # Define the network architecture
+    model = nn.Sequential(
+        nn.Linear(28 * 28, n_units_1),
+        nn.Dropout(p=dropout_1),
+        nn.ReLU(),
+        nn.Linear(n_units_1, n_units_2),
+        nn.Dropout(p=dropout_2),
+        nn.ReLU(),
+        nn.Linear(n_units_2, 10)
+    )
+
+    # Define the SGD optimizer
+    criterion = nn.CrossEntropyLoss()
+    optimizer = torch.optim.Adam(model.parameters(), lr=learning_rate, weight_decay=wd)
+
+    curr_best = None
+    for epoch in range(args.epochs):
+
+        # training
+        model.train()
+
+        for data, target in train_loader:
+            optimizer.zero_grad()
+            output = model(data.view(batch_size, -1))
+            loss = criterion(output, target)
+            loss.backward()
+            optimizer.step()
+
+        # validating
+        model.eval()
+        correct = 0
+        total = 0
+        for data, target in valid_loader:
+            output = model(data.view(batch_size, -1))
+            _, predicted = torch.max(output.data, 1)
+            total += target.size(0)
+            correct += (predicted == target).sum().item()
+
+        acc = correct / total
+
+        # Every time we improve in terms of validation accuracy, we save the network parameters.
+        if curr_best is None or acc > curr_best:
+            curr_best = acc
+            print("checkpoint model")
+            torch.save(model.state_dict(), 'model_checkpoint')
+
+        # We also report the hyperparameter configuration back to AutoGluon,
+        # so we can process it later in the callback function, for example for visualization.
+        config = dict()
+        config["n_units_1"] = n_units_1
+        config["n_units_2"] = n_units_2
+        config["dropout_1"] = dropout_1
+        config["dropout_2"] = dropout_2
+        config["batch_size"] = batch_size
+        config["learning_rate"] = learning_rate
+        config["wd"] = wd
```

Comment:
This is not passed back to reporter, right?

Reply:
you are right, I will fix it